### PR TITLE
use systemd-networkd everywhere

### PR DESCRIPTION
**Pull request by zimbatm**
Stop using piles of bash scripts?

### CODE REVIEW

File: profiles/common.nix [hunk 1]
```diff
@@ -14,6 +14,10 @@
   # Allow PMTU / DHCP
   networking.firewall.allowPing = true;
 
+  # Use networkd instead of the pile of shell scripts
+  networking.useNetworkd = true;
+  networking.useDHCP = false;
```
Comment (Mic92) [inline profiles/common.nix:19]: Since this is also a desktop profile, how does this interact with network-manager?

Comment (zimbatm) [reply profiles/common.nix:19]: I will try it and report back

Comment (zimbatm) [reply profiles/common.nix:19]: Seems to be working fine. NetworkManager is running side-by-side and it still used to connect to the wifi, acquire DHCP, ...

Comment (Mic92) [inline profiles/common.nix:19]: Could be dangerous to overwrite defaults that people are already used to.

```suggestion
  networking.useNetworkd = lib.mkDefault true;
  networking.useDHCP = lib.mkDefault false;
```

**Pull request merged**